test/topology/util.py

```diff
@@ -327,3 +330,62 @@ async def trigger_snapshot(manager, server: ServerInfo) -> None:
     host = cql.cluster.metadata.get_host(server.ip_addr)
     await manager.api.client.post(f"/raft/trigger_snapshot/{group0_id}", host=server.ip_addr)
 
+
+
+async def get_coordinator_host_ids(manager: ManagerClient) -> list[str]:
+    """ Get coordinator host id from history
+
+    Select all records with elected coordinator
+    from description column in system.group0_history table and 
+    return list of coordinator host ids, where
+    first element in list is active coordinator
+    """
+    stm = SimpleStatement("select description from system.group0_history "
+                          "where key = 'history' and description LIKE 'Starting new topology coordinator%' ALLOW FILTERING;")
+
+    cql = manager.get_cql()
+    result = await cql.run_async(stm)
+    coordinators_ids = []
+    for row in result:
+        coordinator_host_id = get_uuid_from_str(row.description)
+        if coordinator_host_id:
+            coordinators_ids.append(coordinator_host_id)
+        continue
+    assert len(coordinators_ids) > 0, f"No coordinator ids {coordinators_ids} were found"
+    return coordinators_ids
+
+
+async def get_coordinator_host(manager: ManagerClient) -> ServerInfo:
+    """Get coordinator ServerInfo"""
+
+    coordinator_host_id = (await get_coordinator_host_ids(manager))[0]
+    server_id_maps = {await manager.get_host_id(srv.server_id):srv for srv in await manager.running_servers()}
+    coordinator_host = server_id_maps.get(coordinator_host_id, None)
+    assert coordinator_host, \
+        f"Node with host id {coordinator_host_id} was not found in cluster host ids {list(server_id_maps.keys())}"
+    return coordinator_host
+
+
+def get_uuid_from_str(string: str) -> str:
+    """Search uuid in string"""
+    uuid = ""
+    if match := UUID_REGEX.search(string):
+        uuid = match.group(1)
+    return uuid
+
+
+async def wait_new_coordinator_elected(manager: ManagerClient, expected_num_of_elections: int, deadline: float) -> None:
+    """Wait new coordinator to be elected
+
+    Wait while the table 'system.group0_history' will have a number of lines 
+    with the 'new topology coordinator' equal to the expected_num_of_elections number,
+    and the latest host_id coordinator differs from the previous one.
+    """
+    async def new_coordinator_elected():
+        coordinators_ids = await get_coordinator_host_ids(manager)
+        if len(coordinators_ids) == expected_num_of_elections \
+            and coordinators_ids[0] != coordinators_ids[1]:
+            return True
+        logger.error("New coordinator was not elected %s", coordinators_ids)
```

Comment:
This is not an error. It's only an error if we timeout. But we may need to wait for some time until coordinator is elected which is expected.

So you could make it `warn` or `info`.

Reply:
set as warning